### DISCUSSION TITLE
Communication hub 2 computer workaround

### DISCUSSION THREAD

**Discussion by AnToHu0** [pybricks/support#1384] [Q&A] [answered]
Hi everyone! I'm trying to establish text connection between hub and computer like in the official example.
I've already implemented a simple echo server on a hub and tryig to communicate with it. Here's the hub's code:

```
from pybricks.tools import wait

from usys import stdin, stdout
from uselect import poll

keyboard = poll()
keyboard.register(stdin)

while True:
    cmd = stdin.readline()
    stdout.write(cmd)
    if cmd == b"bye":
        break
```
It works perfect inside the pybricks ide, but when I try to send data from my node js program using noble it answers very strange. Here's the example of communication log:
```
Sent: Hello, World!
Received: @
Received: @
Sent: Hello, World!
Received: @
Received: @
...
```
On every hello world message it answers with exact two @ symbols in different on-data events each. What am I doing wrong?

The node js code is here: [https://pastebin.com/vaw1UmzE](https://pastebin.com/vaw1UmzE) Please don't blame me for it - it just a chatGPT assisted workaround at the moment.

Maybe it`s because readline and i should send some kind of line termination symbol at the end of each message? I tried \n and \r\n but result is the same.

**Comment by AnToHu0** [comment, marked as answer]
Hah! I figured it out. I forgot about prepend bytes for stdin and stdout. Well python program was ok i`ve just changed a bit the node js code to handle those bytes and all goes work!

**Reply by laurensvalk** [reply]
Glad it worked!
